rcl/src/rcl/client.c

```diff
@@ -288,16 +288,18 @@ rcl_take_response(
   return RCL_RET_OK;
 }
 
-bool rcl_client_is_valid(const rcl_client_t * client)
+bool rcl_client_is_valid(const rcl_client_t * client, const rcl_allocator_t * error_msg_allocator)
 {
   const rcl_client_options_t * options;
-  RCL_CHECK_ARGUMENT_FOR_NULL(
-    client, false, rcl_get_default_allocator());
-  options = _client_get_options(client);
+  const rcl_allocator_t alloc =
+    error_msg_allocator ? *error_msg_allocator : rcl_get_default_allocator();
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&alloc, "error msg allocator is invalid", return false);
+  RCL_CHECK_ARGUMENT_FOR_NULL(client, false, alloc);
   RCL_CHECK_FOR_NULL_WITH_MSG(
-    options, "client's options pointer is invalid", return false, rcl_get_default_allocator());
+    client->impl, "client's rmw implementation is invalid", return false, alloc);
+  options = _client_get_options(client);
   RCL_CHECK_FOR_NULL_WITH_MSG(
-    client->impl, "client's rmw implementation is invalid", return false, options->allocator);
+    options, "client's options pointer is invalid", return false, alloc);
```

Comment:
All the other functions check the x->impl->rmw_handle, I think the client should do the same